resourcemanager/pooltask/task_manager.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pooltask
+
+import (
+	"container/list"
+	"sync"
+	"time"
+
+	"go.uber.org/atomic"
+)
+
+const shard int = 8
+
+func getShardID(id uint64) uint64 {
```

Comment:
Not get point. Why need shard here?

Reply:
In the future, we need to schedule the task with the task manager while we will add or delete the task in the task manager from the user business. Lock contention is a bit intense so we add the shard lock.